src/EditorFeatures/VisualBasicTest/UseObjectInitializer/UseObjectInitializerTests.vb

```diff
@@ -369,6 +369,108 @@ Class C
         z.y = 2
     End Sub
 End Class
+")
+        End Function
+
+        <WorkItem(23368, "https://github.com/dotnet/roslyn/issues/23368")>
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)>
+        Public Async Function TestWithExplicitImplementedInterfaceMembers1() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"
+class C
+    Sub Bar()
+        Dim c As IExample = [||]New Foo
+        c.Name = String.Empty
+    End Sub
+End Class
+
+Interface IExample
+    Property Name As String
+    Property LastName As String
+End Interface
+
+Class Foo
+    Implements IExample
+
+    Private Property Name As String Implements IExample.Name
+    Public Property LastName As String Implements IExample.LastName
+End Class
+")
+        End Function
+
+        <WorkItem(23368, "https://github.com/dotnet/roslyn/issues/23368")>
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)>
+        Public Async Function TestWithExplicitImplementedInterfaceMembers2() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"
+class C
+    Sub Bar()
+        Dim c As IExample = [||]New Foo
+        c.Name = String.Empty
+        c.LastName = String.Empty
+    End Sub
+End Class
+
+Interface IExample
+    Property Name As String
+    Property LastName As String
+End Interface
+
+Class Foo
+    Implements IExample
+
+    Private Property Name As String Implements IExample.Name
+    Public Property LastName As String Implements IExample.LastName
+End Class
+")
+        End Function
+
+        <WorkItem(23368, "https://github.com/dotnet/roslyn/issues/23368")>
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)>
+        Public Async Function TestWithExplicitImplementedInterfaceMembers3() As Task
+            Await TestInRegularAndScriptAsync(
+"
+class C
+    Sub Bar()
+        Dim c As IExample = [||]New Foo
+        c.LastName = String.Empty
+        c.Name = String.Empty
+    End Sub
+End Class
+
+Interface IExample
+    Property Name As String
+    Property LastName As String
+End Interface
+
+Class Foo
+    Implements IExample
+
+    Private Property Name As String Implements IExample.Name
+    Public Property LastName As String Implements IExample.LastName
+End Class
+",
+"
+class C
+    Sub Bar()
+        Dim c As IExample = New Foo With {
+            .LastName = String.Empty
+        }
+        c.Name = String.Empty
+    End Sub
+End Class
+
+Interface IExample
+    Property Name As String
+    Property LastName As String
+End Interface
+
+Class Foo
+    Implements IExample
+
+    Private Property Name As String Implements IExample.Name
+    Public Property LastName As String Implements IExample.LastName
```

Comment:
can we have a test where these names don't match.  i.e. "Public Property MyLastName Implements IExample.LastName"?